ccl/crosscluster/physical: TestAlterTenantCompleteToLatest failed

ccl/crosscluster/physical.TestAlterTenantCompleteToLatest [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16102107?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16102107?buildTab=artifacts#/) on master @ [89aa5c870293c3fe0db6a5be8ee616e271b508c5](https://github.com/cockroachdb/cockroach/commits/89aa5c870293c3fe0db6a5be8ee616e271b508c5):


```
=== RUN   TestAlterTenantCompleteToLatest
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/a52355e0a9a05347a789d171cedfed59/logTestAlterTenantCompleteToLatest915054443
    test_log_scope.go:81: use -show-logs to present logs inline
    testutils.go:290: error scanning '&{<nil> 0x40057e61b0}': sql: Scan error on column index 0, name "retained_time": unsupported Scan, storing driver.Value type <nil> into type *time.Time
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a52355e0a9a05347a789d171cedfed59/logTestAlterTenantCompleteToLatest915054443
--- FAIL: TestAlterTenantCompleteToLatest (7.39s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterTenantCompleteToLatest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/ccl/crosscluster/replicationtestutils/BUILD.bazel
+++ b/pkg/ccl/crosscluster/replicationtestutils/BUILD.bazel
@@ -53,6 +53,7 @@ go_library(
         "//pkg/util/timeutil",
         "@com_github_cockroachdb_apd_v3//:apd",
         "@com_github_cockroachdb_errors//:errors",
+        "@com_github_lib_pq//:pq",
         "@com_github_stretchr_testify//require",
     ],
 )

--- a/pkg/ccl/crosscluster/replicationtestutils/testutils.go
+++ b/pkg/ccl/crosscluster/replicationtestutils/testutils.go
@@ -47,6 +47,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/retry"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/errors"
+	"github.com/lib/pq"
 	"github.com/stretchr/testify/require"
 )
 
@@ -284,12 +285,13 @@ func (c *TenantStreamingClusters) Cutover(
 	require.LessOrEqual(c.T, protectedTimestamp.GoTime(), cutoverOutput.GoTime())
 
 	// PTS should be less than or equal to retained time as a result of heartbeats.
-	var retainedTime time.Time
+	var retainedTime pq.NullTime
 	c.DestSysSQL.QueryRow(c.T,
 		`SELECT retained_time FROM [SHOW TENANT $1 WITH REPLICATION STATUS]`,
 		c.Args.DestTenantName).Scan(&retainedTime)
-
-	require.LessOrEqual(c.T, protectedTimestamp.GoTime(), retainedTime)
+	if retainedTime.Valid {
+		require.LessOrEqual(c.T, protectedTimestamp.GoTime(), retainedTime.Time)
+	}
 
 	if !async {
 		jobutils.WaitForJobToSucceed(c.T, c.DestSysSQL, jobspb.JobID(ingestionJobID))

